### PR TITLE
Migrate `unknown-mod-stdin`, `issue-68794-textrel-on-minimal-lib`, `raw-dylib-cross-compilation` and `used-cdylib-macos` `run-make` tests to rmake

### PR DESCRIPTION
**Pull request by Oneirical**
Part of #121876 and the associated [Google Summer of Code project](https://blog.rust-lang.org/2024/05/01/gsoc-2024-selected-projects.html).

Seriously needs OSX/Windows try-jobs. If it fails, restore `only-linux` in `textrel-on-minimal-lib` and try again.

try-job: x86_64-mingw
try-job: x86_64-msvc

### CODE REVIEW

**Comment by Oneirical** [comment]
Hey, that's pretty neat.
@rustbot review

**Review by jieyouxu — APPROVED** [review]
Thanks, tests look good, r=me if the new try-jobs come back green.

**Comment by Oneirical** [comment]
It failed on OSX because it was only printing the dynamic table, not all relevant sections like `nm` would do. Changed `--dt` to `--all`.

@rustbot review

**Comment by jieyouxu** [comment]
Thanks!
@bors r+ rollup=iffy

File: tests/run-make/raw-dylib-cross-compilation/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        .arg("-a")
+        .input(rust_lib_name("i686_raw_dylib_test"))
+        .run()
+        .assert_stdout_contains("file format coff-i386")
+        .assert_stdout_not_contains("file format coff-x86-64");
+    // Build as x64 and make sure that we have x64 objects only.
+    rustc()
+        .crate_type("lib")
+        .crate_name("x64_raw_dylib_test")
+        .target("x86-64-pc-windows-gnu")
```
Comment (ChrisDenton) [inline tests/run-make/raw-dylib-cross-compilation/rmake.rs:32]: Typo. Should be `x86_64-pc-windows-gnu`

Comment (Oneirical) [reply tests/run-make/raw-dylib-cross-compilation/rmake.rs]: That's so... silly. Thank you so much!

**Comment by Oneirical** [comment]
Change made - restored ignore directive to the original Makefile one and added it to compiletest. This is still worth a @bors try

**Comment by Oneirical** [comment]
Maybe it's just me, but actually spelling the name of the crate correctly could help. @bors try

**Comment by Oneirical** [comment]
Neat! @rustbot review

**Comment by Kobzol** [comment]
r? @Kobzol

@bors delegate+

@bors rollup=iffy

Feel free to r=me after a rebase.

**Comment by Kobzol** [comment]
Hmm, that *looks* spurious.

@bors retry

**Pull request merged**